src/sentry/integrations/bitbucket_server/integration.py

```diff
@@ -221,7 +221,10 @@ class BitbucketServerIntegration(IntegrationInstallation, RepositoryMixin):
 
     def get_client(self):
         if self.default_identity is None:
-            self.default_identity = self.get_default_identity()
+            try:
+                self.default_identity = self.get_default_identity()
+            except Identity.DoesNotExist:
+                raise
```

Comment:
Instead of raising a `RuntimeError` you should either use `IntegrationError` or some new exception. You'll also still have to catch the exception raised here in the webhook and return an error code other than 500.